opty/direct_collocation.py

```diff
@@ -448,6 +450,35 @@ def plot_constraint_violations(self, vector, axes=None):
         - s : number of unknown time intervals
 
         """
+        def extract_mantissa_exponent(number):
+            """Returns the mantissa and exponent of a number."""
+            mantissa, exponent = np.frexp(number)
+            exponent *= np.log10(2)
+            mantissa *= 10**(exponent % 1)
+            exponent = int(exponent)
+            return mantissa, exponent
+
+        bars_per_plot = 10
+        rotation = -90
+
+        # ensure that len(axes) is correct, raise ValuError otherwise
+        if axes is not None:
+            warner = False
+            for i in range(len(axes.ravel())):
+                if axes.ravel()[i]._sharex is not None:
+                    warner = True
+            if warner == True:
+                print('Set sharex=False or remove, it makes no sense here')
```

Comment:
We typically don't have print statements in library code because their display can't be controlled. Best to avoid warnings and just raise errors.

Reply:
I also don't think we need to warn about things like this. The best approach here is to explain in the docstring what the user must do if they want to pass in their own axes.

Reply:
> I also don't think we need to warn about things like this. The best approach here is to explain in the docstring what the user must do if they want to pass in their own axes.

If sharex = True, the program will not interrupt, but the plots are meaningless. Do you mean, I just let this happen, and explain in the method, that it should be set to False or removed? Should this be mentioned in the opty Documentation?
If YES, I have to find out how to do this.

Reply:
> Do you mean, I just let this happen, and explain in the method, that it should be set to False or removed?

Yes, but I don't think you need to explain anything about sharex. `plt.subplots()` has a very long list of arguments and it is not our job to explain them all here. You simply have to state to the user the minimal information they need to know to pass in a correct axes object.